src/modules/verifiableCredentials/verifiableCredentials.types.ts

```diff
@@ -32,19 +51,45 @@ export interface SignedVerifiableCredential<T> extends VerifiableCredential<T> {
   };
 }
 
-export interface SignVerifiableCredentialOptions {
+export interface ProofOptions {
   verificationMethod?: string;
   proofPurpose?: string;
-  domain?: Partial<TypedDataDomain>;
+  domain?: TypedDataDomain;
 }
 
-export interface VerifyVerifiableCredentialOptions {
-  proofPurpose?: string;
-  verificationMethod?: string;
-}
+export type VerifyVerifiableCredentialOptions = Omit<ProofOptions, 'domain'>;
 
 export interface VerifyVerifiableCredentialResults {
   checks: string[];
   warnings: string[];
   errors: string[];
 }
+
+export interface Presentation {
+  '@context': string[];
+  id: string;
+  type: 'VerifiablePresentation';
+  holder: string;
+  verifiableCredential?: VerifiableCredential<unknown>[];
+}
+
+export interface VerifiablePresentation extends Presentation {
+  proof: {
+    '@context': string | string[];
+    type: string;
+    proofPurpose: string;
+    proofValue: string;
+    verificationMethod: string;
+    created: string;
+    eip712Domain: {
+      domain: TypedDataDomain;
+      messageSchema: {
+        VerifiablePresentation: TypedDataField[];
+        EIP712Domain: TypedDataField[];
+        Proof: TypedDataField[];
+        VerifiableCredential?: TypedDataField[];
+      };
+      primaryType: string;
+    };
+  };
+}
```

Comment:
Instead of our own types, could we import from PEX? I think we'll probably be using PEX in ICL, so I don't think it will add additional weight to the library https://github.com/Sphereon-Opensource/pex/blob/c23a250c8581c76fbefac59fca90b418bbe19c2f/lib/types/SSI.types.ts#L104 
I think PEX doesn't have some of the special eip712Domain details, but perhaps that isn't a problem or can be added via another interface?

Reply:
Okay, if there is an intention to use PEX in ICL in the future i think it's not a problem.